modules/_node-scripts/format/index.mjs

```diff
@@ -3,305 +3,17 @@
-import path from 'path';
-import prettier from 'prettier';
-import stylelint from 'stylelint';
+import preflight from '../preflight/index.mjs';
+import format from './format.mjs';
 
-import {getRootDir} from '../util/constants.mjs';
-import filterChangedFiles from '../util/filterChangedFiles.mjs';
-import {readIgnoreFile} from '../util/readIgnoreFile.mjs';
-import {ID_END, ID_START} from './jsp/getPaddedReplacement.mjs';
-import processJSP from './jsp/processJSP.mjs';
-import {SCRIPTLET_CONTENT} from './jsp/substituteTags.mjs';
-import {BLOCK_CLOSE, BLOCK_OPEN} from './jsp/tagReplacements.mjs';
-import {FILLER_CHAR, SPACE_CHAR, TAB_CHAR} from './jsp/toFiller.mjs';
-import logError from './logError.mjs';
-
-const PRETTIER_IGNORE_FILE = '.prettierignore';
-const ESLINT_IGNORE_FILE = '.eslintignore';
-const GIT_IGNORE_FILE = '.gitignore';
-
-async function getFilesToCheck(rootDir) {
-	const eslintIgnoreFilePath = path.join(rootDir, ESLINT_IGNORE_FILE);
-	const prettierIgnoreFilePath = path.join(rootDir, PRETTIER_IGNORE_FILE);
-	const gitIgnoreFilePath = path.join(rootDir, GIT_IGNORE_FILE);
-
-	const eslintIgnores = readIgnoreFile(eslintIgnoreFilePath);
-	const prettierIgnores = readIgnoreFile(prettierIgnoreFilePath);
-	const gitIgnores = readIgnoreFile(gitIgnoreFilePath);
-
-	return await fg(
-		[
-			'*.{graphql,js,mjs,scss,ts,tsx}',
-			'**/*.{graphql,js,mjs,scss,ts,tsx}',
-			'**/src/**/*.{jsp,jspf}',
-		],
-		{
-			dot: true,
-			ignore: [
-				'**/src/test/**',
-				'**/build_gradle/**',
-				...gitIgnores,
-				...eslintIgnores,
-				...prettierIgnores,
-			].map((ignore) => {
-				if (ignore.startsWith('*') && !ignore.startsWith('**')) {
-					ignore = `**/${ignore}`;
-				}
-
-				if (!ignore.startsWith('*')) {
-					ignore = `**${ignore.startsWith('/') ? '' : '/'}${ignore}`;
-				}
-
-				if (!ignore.endsWith('**') && !ignore.includes('.')) {
-					ignore = `${ignore}${ignore.endsWith('/') ? '' : '/'}**`;
-				}
-
-				return ignore;
-			}),
-		}
-	);
-}
-
-const FALLBACK_FILE_PATH = '__fallback__.js';
-
-export default async function format() {
+export default async function main() {
 	const fix = process.argv[3] !== '--check';
 
-	const rootDir = await getRootDir();
-
-	let filepaths = await getFilesToCheck(rootDir);
-
-	filepaths = await filterChangedFiles(filepaths);
-
-	const [eslintConfig, prettierConfig, stylelintConfig] = await Promise.all([
-		getEslintConfig(rootDir),
-		getPrettierConfig(rootDir),
-		getStylelintConfig(rootDir),
-	]);
-
-	const eslintCLI = new ESLint({
-		baseConfig: eslintConfig,
-		fix: true,
-		ignorePath: path.join(rootDir, ESLINT_IGNORE_FILE),
-	});
-
-	const badFiles = [];
-	const errMessages = [];
-	let checked = 0;
-	let fixed = 0;
-
-	async function formatWithPrettier(input, filepath, configOverride = {}) {
-		return await prettier.format(input, {
-			...prettierConfig,
-			...configOverride,
-			filepath,
-		});
+	if (!fix) {
+		console.log('Running preflight...');
+		await preflight();
```

Comment:
I figure preflight really is only necessary in CI settings. This way when someone runs format locally, we don't run tsc all the time